src/data_watcher.cpp

```diff
@@ -161,31 +166,133 @@ std::optional<std::vector<eventInfo>> DataWatcher::readEvents()
 bool DataWatcher::processEvent(const eventInfo& receivedEventInfo)
 {
     if ((std::get<eventMask>(receivedEventInfo) & IN_CLOSE_WRITE) != 0)
+    {
+        return processCloseWrite(receivedEventInfo);
+    }
+    else if ((std::get<eventMask>(receivedEventInfo) &
+              (IN_CREATE | IN_ISDIR)) != 0)
     {
         /**
-         * Either of the following case occured
-         * Case 1 : Files listed in the config and are already existing has been
-         *          modified
-         * Case 2 : Files got created inside a watching directory
+         * Handle the creation of directories inside a monitoring DIR
          */
-        lg2::debug("Processing an IN_CLOSE_WRITE for {PATH}", "PATH",
-                   _watchDescriptors.at(std::get<wd>(receivedEventInfo)) /
-                       std::get<basename>(receivedEventInfo));
+        return processCreate(receivedEventInfo);
+    }
+    return false;
+}
+
+bool DataWatcher::processCloseWrite(const eventInfo& receivedEventInfo)
```

Comment:
Looks many places, you are trying to use `at` API to get something from map, why cannot get once and use it all places? The same comment is applicable to all commits (from the start).